src/practices/Java/JavaNamingConventionsPractice.ts

```diff
@@ -0,0 +1,46 @@
+import { IPractice } from '../IPractice';
+import { PracticeEvaluationResult, PracticeImpact, ProgrammingLanguage } from '../../model';
+import { DxPractice } from '../DxPracticeDecorator';
+import { PracticeContext } from '../../contexts/practice/PracticeContext';
+
+@DxPractice({
+  id: 'Java.NamingConventions',
+  name: 'Use Java Naming Conventions',
+  impact: PracticeImpact.small,
+  suggestion: 'Java class names should begin capitalized as UpperCamelCase or PascalCase in a regular naming convention.',
+  reportOnlyOnce: true,
+  url: 'https://www.oracle.com/technetwork/java/codeconventions-135099.html',
+})
+export class JavaNamingConventionsPractice implements IPractice {
+  async isApplicable(ctx: PracticeContext): Promise<boolean> {
+    return ctx.projectComponent.language === ProgrammingLanguage.Java;
+  }
+
+  async evaluate(ctx: PracticeContext): Promise<PracticeEvaluationResult> {
+    if (!ctx.fileInspector) {
+      return PracticeEvaluationResult.unknown;
+    }
+
+    const regexDotJava = new RegExp('.java', 'i');
+    const javaFiles = await ctx.fileInspector.scanFor(regexDotJava, '/', { shallow: false });
+
+    if (javaFiles.length === 0) {
+      return PracticeEvaluationResult.unknown;
+    }
+
+    const incorrectFiles = [];
+
+    javaFiles.forEach((file) => {
+      const correctPascalCase = file.baseName.replace(/^\w/, (firstChar) => firstChar.toUpperCase());
+      if (file.baseName !== correctPascalCase) {
+        incorrectFiles.push(file.baseName);
+      }
```

Comment:
What about checking, if the user doesn't use the snake case? I.e if there is "_" in the file name.

Reply:
I have added this consideration. The code is now using a `camelcase` library instead of a regex value to convert a string into a PascalCase naming convention string. I have also adjusted the practice for Kotlin. Please re-review the changes and let me know if anything needs changing.